File: ExternalC2/go-external-c2/beacon_id.go
package externc2

import (
	"fmt"

	"github.com/google/uuid"
)

// BeaconID is  combination of the internal identifier and Cobalt Strike identifier.
type BeaconID struct {
	CobaltStrikeID int
	InternalID     string
}

// ToString returns this as a string.
func (b *BeaconID) ToString() string {
	return fmt.Sprintf("%d_%s", b.CobaltStrikeID, b.InternalID)
}

// NewBeaconID returns a new BeaconID with the InternalID initialized.
func NewBeaconID() (BeaconID, error) {
	var beaconID BeaconID
	u, err := uuid.NewRandom()
	if err != nil {
		return beaconID, err
	}
	beaconID.InternalID = u.String()
	return beaconID, nil
}

File: ExternalC2/ExternalC2/BeaconId.cs
using System;

namespace ExternalC2
{
    /// <summary>
    ///     A combination of the internal identifier and Cobalt Strike identifier
    /// </summary>
    public struct BeaconId
    {
        /// <summary>
        ///     The cobalt strike identifier
        /// </summary>
        public int CobaltStrikeId;

        /// <summary>
        ///     The internal identifier
        /// </summary>
        public Guid InternalId;

        /// <summary>
        ///     Returns a <see cref="string" /> that represents this instance.
        /// </summary>
        /// <returns>
        ///     A <see cref="string" /> that represents this instance.
        /// </returns>
        public override string ToString()
            => $"{CobaltStrikeId}_{InternalId}";
    }
}
